src/view/com/composer/SelectMediaButton.tsx

```diff
@@ -0,0 +1,495 @@
+      /**
+       * We don't care too much about mimeType at this point on native,
+       * since the `processVideo` step later on will convert to `.mp4`.
+       */
+      if (isWeb && !isSupportedVideoMimeType(mimeType)) {
+        errors.add(SelectedAssetError.Unsupported)
+        continue
+      }
+    }
+
+    if (type === 'image') {
+      if (!isSupportedImageMimeType(mimeType)) {
+        errors.add(SelectedAssetError.Unsupported)
+        continue
+      }
+    }
+
+    if (type === 'gif') {
+      if (isNative) {
+        errors.add(SelectedAssetError.NoGifsOnNative)
+        continue
+      }
+    }
+
+    /*
+     * All validations passed, we have an asset!
+     */
+    supportedAssets.push({
+      mimeType,
+      ...asset,
+    })
+  }
+
+  if (selectableAssetType === 'image') {
+    if (supportedAssets.length > selectionCountRemaining) {
+      errors.add(SelectedAssetError.MaxImages)
+      supportedAssets = supportedAssets.slice(0, selectionCountRemaining)
+    }
+  } else if (selectableAssetType === 'video') {
+    if (supportedAssets.length > 1) {
+      errors.add(SelectedAssetError.MaxVideos)
+      supportedAssets = supportedAssets.slice(0, 1)
+    }
+
+    const selectedVideo = supportedAssets[0]
+
+    if (typeof selectedVideo.duration !== 'number') {
+      try {
+        const metadata = await getAdditionalVideoMetadata(selectedVideo)
+        selectedVideo.duration = metadata.duration
+        selectedVideo.width = metadata.width
+        selectedVideo.height = metadata.height
+      } catch (e: any) {
+        logger.error(`processSelectedAssets: failed to get video metadata`, {
+          safeMessage: e.message,
+        })
+        errors.add(SelectedAssetError.Unsupported)
+        supportedAssets = []
+      }
+    } else {
+      /*
+       * The `duration` is in seconds on web, but in milliseconds on
+       * native. We normalize to milliseconds.
+       */
+      if (isWeb) {
+        selectedVideo.duration = selectedVideo.duration * 1000
+      }
+    }
+
+    if (
+      selectedVideo.duration &&
+      selectedVideo.duration > VIDEO_MAX_DURATION_MS
+    ) {
+      errors.add(SelectedAssetError.VideoTooLong)
+      supportedAssets = []
+    }
+  } else if (selectableAssetType === 'gif') {
+    if (supportedAssets.length > 1) {
+      errors.add(SelectedAssetError.MaxGIFs)
+      supportedAssets = supportedAssets.slice(0, 1)
+    }
+  }
+
+  return {
+    type: selectableAssetType!, // set above
+    assets: supportedAssets,
+    errors,
+  }
+}
+
+export function SelectMediaButton({
+  disabled,
+  allowedAssetTypes,
+  selectedAssetsCount,
+  onSelectAssets,
+}: SelectMediaButtonProps) {
+  const {_} = useLingui()
+  const {requestPhotoAccessIfNeeded} = usePhotoLibraryPermission()
+  const {requestVideoAccessIfNeeded} = useVideoLibraryPermission()
+  const sheetWrapper = useSheetWrapper()
+  const t = useTheme()
+
+  const selectionCountRemaining = MAX_IMAGES - selectedAssetsCount
```

Comment:
I like that MAX_IMAGES is a variable in cases later this variable gets updated, we only need to update it once